### ISSUE TITLE
NetIF error if interface has no IP assigned

### ISSUE THREAD

**Issue by Dr-Syn** [hasameli/foghorn#51]
In the case where an interface exists but does not have an IP assigned, netifaces throws a key error when enumerating that interface.

```
 Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/twisted/application/app.py", line 648, in run
    runApp(config)
  File "/usr/local/lib/python2.7/site-packages/twisted/scripts/twistd.py", line 25, in runApp
    _SomeApplicationRunner(config).run()
  File "/usr/local/lib/python2.7/site-packages/twisted/application/app.py", line 379, in run
    self.application = self.createOrGetApplication()
  File "/usr/local/lib/python2.7/site-packages/twisted/application/app.py", line 444, in createOrGetApplication
    application = getApplication(self.config, passphrase)
--- <exception caught here> ---
  File "/usr/local/lib/python2.7/site-packages/twisted/application/app.py", line 455, in getApplication
    application = service.loadApplication(filename, style, passphrase)
  File "/usr/local/lib/python2.7/site-packages/twisted/application/service.py", line 411, in loadApplication
    passphrase)
  File "/usr/local/lib/python2.7/site-packages/twisted/persisted/sob.py", line 223, in loadValueFromFile
    eval(codeObj, d, d)
  File "foghornd.tac", line 81, in <module>
    for item in foghord_service():
  File "foghornd.tac", line 54, in foghord_service
    for addr in netifaces.ifaddresses(iface)[netifaces.AF_INET]:
exceptions.KeyError: 2
```

Setting an IP on the interface fixes this, but there are certain cases [ e.g. promisc interfaces ] where an IP oughtn't be assigned. May want to add a check to skip over those ifaces.

**Comment by zallison** [comment]
This is fixed in: 4b1e700f42189da1400f42836f293a1cf3395618

**Comment by Dr-Syn** [comment]
Ty

On October 26, 2016 4:38:57 PM PDT, Zack Allison notifications@github.com wrote:

> This is fixed in: 4b1e700f42189da1400f42836f293a1cf3395618
> 
> ## 
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/hasameli/foghorn/issues/51#issuecomment-256507895

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.
